### DISCUSSION TITLE
How to turn on the board after the standby (low power mode)?

### DISCUSSION THREAD

**Discussion by artalar** [microsoft/devicescript#675] [Q&A]
Hi! I want to implement a "turn on/turn off" button for battery-saving mode, and I found that the `standby` API is exactly what I need for the "turn off" scenario. However, I can't figure out how to react to the button to turn the board back to normal mode.

Right now, I have the following code:

```ts
import { standby } from "@devicescript/runtime"

const btn = startButton({ pin: gpio(9) })
btn.down.subscribe(async () => {
    await standby(Infinity)
})
```

**Comment by mmoskal** [comment]
We currently do not support setting wake up modes on pins.

If someone wants to submit a PR, there are [ESP32 APIs](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/system/sleep_modes.html#_CPPv433esp_deep_sleep_enable_gpio_wakeup8uint64_t33esp_deepsleep_gpio_wake_up_mode_t) to do it. This would need to be called from impl_gpio.c, through some function added in jd_pins.h.
